Dockerfile

```diff
@@ -1,11 +1,14 @@
-FROM golang:1.20.2-alpine3.16 as builder
+FROM --platform=$BUILDPLATFORM golang:1.20.2-alpine3.16 as builder
+
+ARG TARGETOS
+ARG TARGETARCH
```

Comment:
This allows cross compilation to `TARGETOS` and `TARGETARCH` directly from the host instead of relying on QEMU's arch emulation. The results is a much faster and less buggy/error prone compilation.

Reply:
Just of curiosity, what errors did you face with the qemu emulation here? Was it different for other repos like up & receive-controller?

Reply:
I was facing multiple SEGFAULTs in the Go toolchain. Tried different versions of Go, container engines, and even base images... nothing worked.

In fact, I would love to migrate all of the project to this method at some point. The benefits are awesome.

Reply:
I didn't have issues with other projects back then. I even tried again to compile a multi-arch image of a few of them and they still work fine.  🤷